src/shared/test/integration/inter-frame-communication-test.js

```diff
@@ -0,0 +1,241 @@
+import { delay } from '../../../test-util/wait';
+import { Bridge as BridgeOri } from '../../bridge';
+import { ListenerCollection as ListenerCollectionOri } from '../../listener-collection';
+import { PortFinder as PortFinderOri } from '../../port-finder';
+import { PortProvider as PortProviderOri } from '../../port-provider';
+import { isMessageEqual } from '../../port-util';
+
+describe('PortProvider-PortFinder-Bridge integration', () => {
+  let destroyables;
+
+  // These proxy classes are to keep track of instances that need to be destroyed.
+  class Bridge extends BridgeOri {
+    constructor() {
+      super();
+      destroyables.push(this);
+    }
+  }
+
+  class ListenerCollection extends ListenerCollectionOri {
+    constructor() {
+      super();
+      destroyables.push(this);
+    }
+    destroy() {
+      this.removeAll();
+    }
+  }
+
+  class PortFinder extends PortFinderOri {
+    constructor(...args) {
+      super(...args);
+      destroyables.push(this);
+    }
+  }
+
+  class PortProvider extends PortProviderOri {
+    constructor(...args) {
+      super(...args);
+      destroyables.push(this);
+    }
+  }
+
+  beforeEach(() => {
+    destroyables = [];
+  });
+
+  afterEach(() => {
+    destroyables.forEach(instance => instance.destroy());
+  });
+
+  it('enables the communication between sidebar-host and notebook-sidebar frames', async () => {
+    let done;
+    const promise = new Promise(resolve => (done = resolve));
+
+    // Test this fictional message path:
+    // host -- ping --> sidebar -- ping --> notebook
+    // host <- pong --- sidebar <- pong --- notebook
+
+    // host frame
+    const portProvider = new PortProvider(window.location.origin);
+    const hostBridge = new Bridge();
+
+    // register RPC method *before* creating the channel
```

Comment:
```suggestion
    // Register RPC method *before* creating the channel
```

Same for other comments.